src/chunks/storage/array.rs

```diff
@@ -274,6 +277,50 @@ impl WriteChunk for NdArray {
     }
 }
 
+impl StoragePrune for NdArray {
+    fn simple_prune_storage(&self, toss_indices: &[usize]) -> StorageWrap {
+        let toss_indices: HashSet<usize> = HashSet::from_iter(toss_indices.iter().cloned());
+        let mut keep_indices_all = Vec::new();
+        for idx in 0..self.shape().1 {
+            if !toss_indices.contains(&idx) {
+                keep_indices_all.push(idx);
+            }
+        }
+        NdArray::new(self.inner.select(Axis(0), &keep_indices_all).to_owned()).into()
+    }
+
+    fn most_similar(
+        &self,
+        keep_indices: &[usize],
+        toss_indices: &[usize],
+        batch_size: usize,
+    ) -> Array1<usize> {
+        let toss = toss_indices.len();
+        let mut remap_indices = Array1::zeros(toss);
+        let keep_embeds = self.inner.select(Axis(0), keep_indices);
+        let keep_embeds_t = &keep_embeds.t();
+        let toss_embeds = self.inner.select(Axis(0), toss_indices);
+        for n in (0..toss).step_by(batch_size) {
+            let mut offset = n + batch_size;
+            if offset > toss {
+                offset = toss;
+            }
+            let batch = toss_embeds.slice(s![n..offset, ..]);
```

Comment:
```clippy``` complains here, but this seems to be a [bug](https://github.com/rust-ndarray/ndarray/issues/474)

Reply:
You can add this above the line to silence clippy `#[allow(clippy::deref_addrof)]`, I think we have used that at other parts as well, tho I never looked deeper into why that happens.

Reply:
Btw, this is another case where clippy complained on my laptop and not on CI